Student-Unity/src/main/java/csed/swe/studentunity/API/TaskAPI.java

```diff
@@ -0,0 +1,34 @@
+package csed.swe.studentunity.API;
+
+import csed.swe.studentunity.Logic.TaskService;
+import csed.swe.studentunity.model.Task;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/tasks")
+public class TaskAPI {
+
+    private final TaskService taskService;
+
+    public TaskAPI(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @PostMapping("/addTask")
+    public ResponseEntity<Task> addTask(@RequestBody Task task) {
```

Comment:
You said the admin has the ability to do those operations but where is the check that the request is sent by the admin?
I advise you to use an interceptor to check that the request is sent by the admin.

Reply:
really i forgot to check that only the admin who can add a task
i will use the interceptor